### ISSUE TITLE
Docker can't connect to solr: `ConnectionError: HTTPConnectionPool(host='solr', port=8983)`

### ISSUE THREAD

**Issue by scottbarnes** [internetarchive/openlibrary#8948]
### Problem

TLDR: if you get this error, try `docker network connect openlibrary_webnet openlibrary-web-1`.

Sometimes when I (and others) rebuild the docker containers in the local development environment, out of nowhere I get the following error (provided in text to make it easier to search, with line breaks added):
```
/openlibrary/openlibrary/templates/home/index.html: error in processing
 template: ConnectionError: HTTPConnectionPool(host='solr', port=8983): 
Max retries exceeded with url: /solr/openlibrary/select (Caused by 
NameResolutionError("<urllib3.connection.HTTPConnection object at 
0x77a95c4e7f90>: Failed to resolve 'solr' ([Errno -2] Name or service 
not known)")) (falling back to default template)
```

#### Evidence / Screenshot
![image](https://github.com/internetarchive/openlibrary/assets/26524678/f4f54b93-5de7-4c65-97f5-32c88500de12)

One cause seems to be that the containers become disconnected from `openlibrary_webnet` (though conceivably this could happen with `openlibrary_dbnet` too):
```sh
❯ docker container inspect --format '{{.NetworkSettings.Networks}}' openlibrary-web-1
map[openlibrary_dbnet:0xc00037c1c0]
```

However, if one looks at `compose.yaml`, `openlibrary-web-1` should be connected to both `webnet` and `dbnet`:
```yaml
version: "3.8"
services:
  web:
    image: "${OLIMAGE:-oldev:latest}"
    environment:
      - OL_CONFIG=${OL_CONFIG:-/openlibrary/conf/openlibrary.yml}
      - GUNICORN_OPTS=${GUNICORN_OPTS:- --reload --workers 4 --timeout 180}
    command: docker/ol-web-start.sh
    ports:
      - ${WEB_PORT:-8080}:8080
    networks:
      - webnet
      - dbnet
```
Running `docker inspect openlibrary_webnet` shows that `openlibrary-web-1` is *not* there (output omitted).

A fix seems to be:
```
❯ docker network connect openlibrary_webnet openlibrary-web-1

❯ docker container inspect --format '{{.NetworkSettings.Networks}}' openlibrary-web-1
map[openlibrary_dbnet:0xc00016c460 openlibrary_webnet:0xc00016c540]
```
No restart is required. If `webnet` no longer exists, recreating it _may_ fix things: `docker network create openlibrary_webnet`. This is, however, speculation.

**Comment by scottbarnes** [comment]
I just opened this in case people search `git` for it.
